Implementation/FindMyBLEDevice/FindMyBLEDevice/Views/ItemDetailPage.xaml

```diff
@@ -17,7 +17,24 @@
 
     <StackLayout Spacing="20" Padding="15">
         <Label Text="User Label:" FontSize="Medium" TextColor="Black" />
-        <Label Text="{Binding Device.UserLabel}" FontSize="Small" TextColor="Black"/>
+
+        <StackLayout Spacing="0">
+            <Entry x:Name="UserLabel" Text="{Binding UserLabel, Mode=TwoWay}" Placeholder="{Binding Device.UserLabel}" FontSize="Small" TextColor="Black" TextChanged="UserLabel_TextChanged"/>
+            <Label Text="For renaming the device, edit the label in the entry" FontSize="Small" TextColor="Gray" Padding="0,0,0,0" />
+            
+        </StackLayout>
+
+        <Button
+            x:Name="RenameButton"
+            Text="Rename"
+            BackgroundColor="White"
+            TextColor="Black"
+            BorderColor="Black"
+            BorderWidth="1"
+            Margin="3"
+            Command="{Binding RenameButtonTapped}"
+            IsVisible="{Binding UserLabelEdited}"
```

Comment:
```suggestion
            IsEnabled="{Binding UserLabelEdited}"
```
How about this?

Reply:
new ux looks fine to me, honestly no strong opinion regarding disabled / not shown button from my side

Reply:
It's a proposal, but the decision is yours